addons/account/models/account_move.py

```diff
@@ -3591,7 +3561,7 @@ def button_draft(self):
                 # so we also check tax_cash_basis_origin_move_id, which stays unchanged
                 # (we need both, as tax_cash_basis_origin_move_id did not exist in older versions).
                 raise UserError(_('You cannot reset to draft a tax cash basis journal entry.'))
-            if move.restrict_mode_hash_table and move.state == 'posted':
+            if move.restrict_mode_hash_table and move.state == 'posted' and move.inalterable_hash:
```

Comment:
The tour failed because here https://github.com/odoo/odoo/pull/84830/files#diff-94435211de5337f34a7e6ee46ec6c554b0554a5e500939a1d54e83c8872df21dR29
we have an inconsistent state where moves are posted, and in restrict mode, but not hashed anymore.
So now we'll only check the hash (which makes more sense too)